src/order/category/NonemptyFinLinOrd.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import data.fintype.sort
+import data.fin
+import order.category.LinearOrder
+
+/-! # Nonempty finite linear orders
+
+Nonempty finite linear orders form the index category for simplicial objects.
+-/
+
+universe variables u v
+
+open category_theory
+
+set_option old_structure_cmd true
+
+section prio
+
+-- see note [default priority]
+set_option default_priority 100
+
+/-- A typeclass for nonempty finite linear orders. -/
+class nonempty_fin_lin_ord (α : Type*) extends fintype α, decidable_linear_order α, order_bot α, order_top α.
+
+end prio
+
+instance punit.nonempty_fin_lin_ord : nonempty_fin_lin_ord punit :=
+by { refine_struct
+{ .. punit.decidable_linear_ordered_cancel_add_comm_monoid,
+  .. punit.fintype };
+{ intros, exact punit.star <|> exact dec_trivial }, }
```

Comment:
Just use a `begin ... end` block? Using `;` and `<|>` is less clear than just using a multiline proof with braces to focus on goals.

Reply:
Fixed